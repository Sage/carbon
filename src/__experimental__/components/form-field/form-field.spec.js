import React from 'react';
import TestRenderer from 'react-test-renderer';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import FormField from '.';
import FormFieldStyle from './form-field.style';
import classicTheme from '../../../style/themes/classic';
import Label from '../label/label.component';

function render(props, renderer = shallow) {
  return renderer(
    <FormField { ...props }><input /></FormField>
  );
}

describe('FormField', () => {
  describe('default', () => {
    it('renders children with styling', () => {
      expect(render({}, TestRenderer.create)).toMatchSnapshot();
    });
  });

  describe('with a label', () => {
    it('renders the label component above the childen', () => {
      expect(render({
        label: 'Name',
        labelAlign: 'left',
        labelHelp: 'Help me!',
        labelInline: true,
        labelWidth: 20,
        size: 'small'
      }).children()).toMatchSnapshot();
    });

    it('passes the id to the Label htmlFor prop', () => {
      const comp = render({
        id: 'foo',
        name: 'foo',
        label: 'Name'
      });
      expect(comp.find(Label).props().htmlFor).toEqual('foo');
    });
  });

  describe('with fieldHelp', () => {
    describe('default', () => {
      it('renders the FieldHelp component below the childen', () => {
        expect(render({
          fieldHelp: 'Help me!',
          labelInline: true,
          labelWidth: 20
        }).children()).toMatchSnapshot();
      });
    });

    describe('and fieldHelpInline=true', () => {
      it('renders the FieldHelp component below the childen', () => {
        expect(render({
          fieldHelp: 'Help me!',
          fieldHelpInline: true,
          labelInline: true,
          labelWidth: 20
        }).children()).toMatchSnapshot();
      });
    });
  });

  describe('classic theme', () => {
    it('adds custom margin top', () => {
      const wrapper = renderFormFieldStyle({
        theme: classicTheme
      });
      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    describe('when inline', () => {
      it('renders the FieldHelp component below the childen', () => {
        const wrapper = renderFormFieldStyle({
          inline: true
        });

        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});

function renderFormFieldStyle(props) {
  return TestRenderer.create(<FormFieldStyle { ...props } />);
}
