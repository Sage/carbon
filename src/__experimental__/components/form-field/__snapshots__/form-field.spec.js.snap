// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField classic theme adds custom margin top 1`] = `
.c2 + .c0 {
  margin-top: 16px;
}

.c1 + .c0 {
  margin-top: 10px;
}

<div
  className="c0 c1"
/>
`;

exports[`FormField classic theme when inline renders the FieldHelp component below the childen 1`] = `
.c1 + .c0 {
  margin-top: 16px;
}

.c2 + .c0 {
  margin-top: 10px;
}

<div
  className="c0 c1"
/>
`;

exports[`FormField default renders children with styling 1`] = `
.c1 + .c0 {
  margin-top: 16px;
}

.c2 {
  display: block;
}

<div
  className="c0 c1"
>
  <div
    className="c2"
  >
    <input />
  </div>
</div>
`;

exports[`FormField with a label renders the label component above the childen 1`] = `
<styled.div
  inline={true}
>
  <Memo(Label)
    align="left"
    help="Help me!"
    inline={true}
    inputSize="small"
    width={20}
  >
    Name
  </Memo(Label)>
  <input />
</styled.div>
`;

exports[`FormField with fieldHelp and fieldHelpInline=true renders the FieldHelp component below the childen 1`] = `
<styled.div
  inline={true}
>
  <FieldHelp
    labelInline={true}
    labelWidth={20}
  >
    Help me!
  </FieldHelp>
  <input />
</styled.div>
`;

exports[`FormField with fieldHelp default renders the FieldHelp component below the childen 1`] = `
Array [
  <styled.div
    inline={true}
  >
    <input />
  </styled.div>,
  <FieldHelp
    labelInline={true}
    labelWidth={20}
  >
    Help me!
  </FieldHelp>,
]
`;
